<launch>
  <arg name="map_name" default="bme_map"
    doc="Name of the map yaml file without extension, contained in the assets folder." />
  <arg name="bag" default="bme_map_take"
    doc="Name of the rosbag file in the assets/bags folder. First number is the sampling rate (e.g., 5hz), 
  then the odometry noise (e.g., 1e-4 m), then the lidar std (e.g., 1e-2 m)" />
  <arg name="mc" default="1" doc="Monte--Carlo number, only used for naming the result file." />
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]" />
  <arg name="info" default="" doc="Any additional info." />
  <arg name="result_export_file_name" default="amclp_$(arg info)_mc$(arg mc)_$(arg bag)"
    doc="File is placed inside the assets/results directory. " />


  <param name="/use_sim_time" value="true" />
  <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote_rtkgps.launch">
    <arg name="model" value="$(arg model)" />
  </include>

  <node name="map_server" pkg="map_server" type="map_server" output="screen"
    args="$(find dhf_loc)/assets/$(arg map_name).yaml" />

  <include file="$(find turtlebot3_navigation)/launch/amcl.launch">
    <!-- The AMCL+ algorithm in the paper is when selective_resampling=true and update_min=0 -->
    <!-- Otherwise, delete these args. -->
    <arg name="selective_resampling" value="true" />
    <arg name="update_min" value="0" />
    <arg name="initial_pose_x" default="0" />
    <arg name="initial_pose_y" default="0" />
    <arg name="initial_pose_a" default="0" />
  </include>

  <node name="rosbag" pkg="rosbag" type="play" output="screen" required="true"
    args="-q --clock -d 10 $(find dhf_loc)/assets/bags/$(arg bag).bag " />

  <node name="robot_pose_publisher" pkg="robot_pose_publisher" type="robot_pose_publisher">
    <param name="is_stamped" value="true" />
    <param name="publish_frequency" value="5" />
  </node>

  <node pkg="dhf_loc" type="localization_result_exporter_node.py"
    name="localization_result_exporter_node"
    output="screen">
    <param name="export_file_name" value="$(arg result_export_file_name)" />
    <param name="truth_topic" value="/odometry/filtered" />
  </node>


  <!-- <node pkg="rviz" type="rviz" name="rviz"
    args="-d $(find dhf_loc)/rviz/real.rviz" /> -->

  <include file="$(find robot_localization)/launch/navsat_transform_template.launch" />
  <include file="$(find robot_localization)/launch/ekf_template.launch" />

</launch>